kong/plugins/prometheus/exporter.lua

```diff
@@ -31,6 +31,58 @@ package.loaded['prometheus_resty_counter'] = require("resty.counter")
 local kong_subsystem = ngx.config.subsystem
 local http_subsystem = kong_subsystem == "http"
 
+
+-- should we introduce a way to know if a plugin is configured or not?
+local is_prometheus_enabled, register_event do
+  local plugin_name = "prometheus"
+  local cache_key = "prometheus:enabled"
+
+
+  local function is_prometheus_enabled_fetch()
+    for plugin, err in kong.db.plugins:each() do
+      if err then
+        kong.log.crit("could not obtain list of plugins: ", err)
+        return nil, err
+      end
+
+      if plugin.name == plugin_name and plugin.enabled then
+        return true
+      end
+    end
+    return false
+  end
+
+
+  function is_prometheus_enabled()
+    local enabled, err = kong.cache:get(cache_key, nil, is_prometheus_enabled_fetch)
+
+    if err then
+      error("error when checking if prometheus enabled: " .. err)
+    end
+
+    return enabled
+  end
+
+
+  -- invalidate cache when a plugin is added/removed/updated
+  function register_event()
+    local worker_events = kong.worker_events
+    if kong.configuration.database == "off" then
+      worker_events.register(function(data)
+        if data.entity.name == plugin_name then
+          kong.db:invalidate(cache_key)
+        end
+      end, "crud", "plugins")
+
+    else
+      worker_events.register(function()
+        kong.db:invalidate(cache_key)
+      end, "declarative", "flip_config")
```

Comment:
It seems that now there is no event `flip_config`, could you check it again?

Reply:
Do we have any (developer) docs for me to understand the new inc-sync?